utils/zerovec/src/ule/string.rs

```diff
@@ -0,0 +1,32 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::ule::*;
+use std::str;
+
+impl AsVarULE for String {
+    type VarULE = str;
+    fn as_unaligned(&self) -> &str {
+        &self
+    }
+    fn from_unaligned(unaligned: &str) -> Self {
+        unaligned.into()
```

Comment:
No, when you get a value from VarZeroVec you get `&str` directly. This function is only used when creating owned versions of a type.

VarZeroVec works differently from ZeroVec, ZeroVec returns `T`, `VarZeroVec` returns `&T::VarULE`. `VarZeroVec` can't return `&T` because there is nothing that maps to that type in memory, and it can't return an owned `T` because that would require allocation. So it returns a reference to the unsized `T::VarULE`.